oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/ServerOAuth2AuthorizedClientManager.java

```diff
@@ -13,11 +13,10 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.springframework.security.oauth2.client.web.server;
```

Comment:
I debated on placing `ServerOAuth2AuthorizedClientManager` in the `*.oauth2.client.server` package but was concerned on a possible package tangle since it referenced `OAuth2AuthorizeRequest` in the package one-level up. However, I just confirmed that this would not introduce a package tangle so I think we could consider moving it.

@rwinch What are your thoughts. Should we move `ServerOAuth2AuthorizedClientManager` to the (new) `*.oauth2.client.server` package?

Reply:
> A (very) quick search of the codebase doesn't reveal any Reactive interfaces that are not in a server or reactive package.

Take a look at the package `org.springframework.security.oauth2.client.endpoint` and you will notice `ReactiveOAuth2AccessTokenResponseClient` and 3x impl's.

In `org.springframework.security.oauth2.client.authentication` you will see `OAuth2LoginReactiveAuthenticationManager` and `OAuth2AuthorizationCodeReactiveAuthenticationManager`

In `org.springframework.security.oauth2.client.oidc.userinfo` there is `OidcReactiveOAuth2UserService`

In `org.springframework.security.oauth2.client.userinfo` there is `ReactiveOAuth2UserService`

etc...

I agree there seems to be some inconsistencies here. I still leaning towards leaving `ServerOAuth2AuthorizedClientManager` in the same package as `OAuth2AuthorizedClientManager` which both share `OAuth2AuthorizeRequest` (also in the base package).

Thoughts?

Reply:
Actually, I see what the problem is. Now that we removed `ServerOAuth2AuthorizeRequest` which was used by `ServerOAuth2AuthorizedClientManager` and it depended on `ServerWebExchange`, the `Server*` prefix and `server` package makes sense. But now that `ServerWebExchange` is not dependent since we're reusing `OAuth2AuthorizeRequest`, we need to rename the interface to `ReactiveOAuth2AuthorizedClientManager`.